tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/union/WorkerUnionBase.java

```diff
@@ -43,4 +43,15 @@ public void addTask(ParallelTask<T> task) {
 		}
 		tasks.add(task);
 	}
+
+	@Override
+	public void handleClose() throws QueryEvaluationException {
+		try {
+			for (ParallelTask<T> task : tasks) {
+				task.close();
+			}
+		} finally {
+			super.handleClose();
+		}
+	}
```

Comment:
We now try to close all the tasks, since it could be that we have been closed before all the tasks were completed (or even started). There is obviously no guarantee that started tasks can be stopped, so it's all best effort.